http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/RuntimeServiceInterp.scala

```diff
@@ -903,6 +907,14 @@ object RuntimeServiceInterp {
       disk <- diskOpt match {
         case Some(pd) =>
           for {
+            _ <- if (pd.zone == targetZone) F.unit
+            else
+              F.raiseError(
```

Comment:
Should this be an error? Wondering if it should create a new disk in this case.

E.g. instead of `persistentDiskQuery.getActiveByName` we could have `persistentDiskQuery.getActiveByNameAndZone` maybe..

Reply:
Not sure which is better, need to think about it..

Reply:
in terms of persistence, if the UI sends a create runtime request and we try to attach an existing disk to a new runtime but the disk and runtime are in different zones - if we create a new disk then we lose persistence, I think? So maybe error-ing out is best if the zones don't match

Reply:
Yeah I can see that.. also `zone` is not in the `PERSISTENT_DISK` unique key so it wouldn't work anyway to create a new disk with the same project/name in a different zone. So erroring feels right.

How about just tweaking the error message:

"Existing disk ${googleProject.value}/${pd.name.value} is in zone ${pd.zone.value}, which cannot be attached to a runtime in zone ${targetZone.value}. Please create a new disk."

The UI might need a way to actually allow users to create a new disk if they get in this situation..

Reply:
That's a good point - we could also add to the error message `... Please create a new disk or create your runtime in zone ${pd.zone.value}.`

Reply:
Updated it to
```
                  s"existing disk ${pd.projectNameString} is in zone ${pd.zone.value}, and cannot be attached to a runtime in zone ${targetZone.value}. Please create a your runtime in zone ${pd.zone.value} if you'd like to use this disk; or opt to use a new disk",
```